core/src/impl/Kokkos_Core.cpp

```diff
@@ -282,11 +295,212 @@ void initialize_profiling(const InitArguments& args) {
     std::exit(EXIT_SUCCESS);
   }
   Kokkos::Tools::parseArgs(args.tool_args);
+  for (const auto& category_value : Kokkos::Impl::metadata_map) {
+    for (const auto& key_value : category_value.second) {
+      Kokkos::Tools::declareMetadata(key_value.first, key_value.second);
+    }
+  }
 }
 
+std::string version_string_from_int(int version_number) {
+  std::stringstream str_builder;
+  str_builder << version_number / 10000 << "." << (version_number % 10000) / 100
+              << "." << version_number % 100;
+  return str_builder.str();
+}
+#define KOKKOS_METADATA_QUOTE_HELPER(x) #x
+#define KOKKOS_METADATA_QUOTE(x) KOKKOS_METADATA_QUOTE_HELPER(x)
```

Comment:
Unused now that we pulled the git commit hash out

Reply:
7230f06